### PR TITLE
Resync after ImportPrivKey and NeutrinoDB corruption fixes

### PR DESCRIPTION
**Pull request by aldebap**
The following were added/changed/fixed:

. start resync from block 1 instead of genesis after importing a private key;
. fix bug that was corrupting NeutrinoDB;
. fix segmentation fault on RollbackLastBlock();
. fix some tests regarding the changes in RollbackLastBlock();
. I kept some debug traces used during the debugging of the test code;

### CODE REVIEW

File: pktwallet/wallet/wallet.go [hunk 1]
```diff
@@ -831,6 +830,10 @@ out:
 // be locked if the passphrase is incorrect or any other error occurs during the
 // unlock.
 func (w *Wallet) Unlock(passphrase []byte, lock <-chan time.Time) er.R {
+	log.Infof("Unlock() [1]")
+	err_ := er.Errorf("Forged error to get the stack trace")
+	log.Infof("Unlock() [1] stack trace: %v", err_)
```
Comment (cjdelisle) [inline pktwallet/wallet/wallet.go:835]: Wanna remove this before merging ?

File: pktwallet/wallet/wallet.go [hunk 2]
```diff
@@ -1915,10 +1919,28 @@ func (w *Wallet) ImportPrivateKey(scope waddrmgr.KeyScope, wif *btcutil.WIF,
...
+		log.Debugf("ImportPrivateKey() [3] second block timestamp: %v", secondBlockTimestamp)
+
 		bs = &waddrmgr.BlockStamp{
-			Hash:      *w.chainParams.GenesisHash,
-			Height:    0,
-			Timestamp: genesis.Block(w.chainParams.GenesisHash).Header.Timestamp,
+			Hash:      *secondBlockHash,
+			Height:    int32(secondBlockIndex),
+			Timestamp: secondBlockTimestamp,
 		}
```
Comment (cjdelisle) [inline pktwallet/wallet/wallet.go:1944]: What about if someone manually invokes a `resync 0` ? We should handle that as well.

File: pktwallet/wallet/wallet.go (hunk 1 above)
Comment (aldebap) [reply pktwallet/wallet/wallet.go]: Fixed in the last commit: https://github.com/pkt-cash/pktd/pull/344/commits/43861d98a90a5ebbd3db48a03cb657708ec3bbd0

File: pktwallet/wallet/wallet.go (hunk 2 above)
Comment (aldebap) [reply pktwallet/wallet/wallet.go:1935]: Fixed in the last commit: https://github.com/pkt-cash/pktd/pull/344/commits/43861d98a90a5ebbd3db48a03cb657708ec3bbd0

**Pull request merged**